### ISSUE TITLE
[a11y]: FileUploaderDropContainer's input needs to be moved outside label

### ISSUE THREAD

**Issue by TannerS** [carbon-design-system/carbon#10443]
### Package

carbon-components-react

### Package version

v10.50.0

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker

### Description

the FileUploaderDropContainer has it's input within it's label which sparks a a11y issue 

`3.3.2 Labels or Instructions: Text input is nested in label such that input precedes the label text`

I think I fix for this is moving it outside the label 

<img width="1677" alt="image" src="https://user-images.githubusercontent.com/8866319/149418455-a9c13e5d-5bbe-4618-8274-333dd16d3c23.png">

<img width="1680" alt="image" src="https://user-images.githubusercontent.com/8866319/149418480-72adf362-d38e-4e5d-832a-2b60432f6709.png">

### WCAG 2.1 Violation

3.3.2 Labels or Instructions

### CodeSandbox example

https://react.carbondesignsystem.com/?path=/story/components-fileuploader--file-uploader-drop-container

### Steps to reproduce

You can view it by ruinning the IBM Equal Access Accessibility Checker in the storybook for the FileUploaderDropContainer

https://react.carbondesignsystem.com/?path=/story/components-fileuploader--file-uploader-drop-container

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
